tools/esdlcmd/README.md

```diff
@@ -0,0 +1,424 @@
+
+### Manifest File
+
+A simplified example showing the format of a manifest file is below.
+
+```xml
+<em:Manifest xmlns:em="urn:hpcc:esdl:manifest">
+    <em:ServiceBinding service="WsFoobar" id="WsFoobar_desdl_binding">
+        <Methods>
+            <Scripts>
+                <em:Include file="WsFoobar-request-prep.xml"/>
+                <em:Include file="WsFoobar-logging-prep.xml"/>
+            </Scripts>
+            <Method name="FoobarSearch" url="127.0.0.1:8888">
+                <Scripts>
+                    <em:Include file="FoobarSearch-scripts.xml"/>
+                </Scripts>
+            </Method>
+        </Methods>
+        <LoggingManager>
+            <LogAgent transformSource="local" name="main-logging">
+                <LogDataXPath>
+                    <LogInfo name="PreparedData" xsl="log-prep">
+                </LogDataXPath>
+                <XSL>
+                    <Transform name="log-prep">
+                        <em:Include file="log-prep.xslt">
+                    </Transform>
+                </XSL>
+            </LogAgent>
+        </LoggingManager>
+    </em:ServiceBinding>
+    <em:ServiceDefinition>
+        <em:Include file="WsFoobar.ecm"/>
+    </em:ServiceDefinition>
+</em:Manifest>
+```
+
+The manifest nodes must use the namespace "urn:hpcc:esdl:manifest", and includes elements configuring asingle service.
+
+A `<em:ServiceBinding>` element contains the binding configuration for the service, including:
+
+1. Enumeration of all methods in the service.
+2. Configuration of the back-end of each method and any method-specific configuration.
+3. ESDL scripts defined in the scope of the service.
+4. ESDL scripts defined in the scope of each method.
+5. Logging configuration that may specify XSLT transforms a LogAgent needs.
+
+ServiceBinding Element Attributes
+
+Attribute   | Required? | Value  | Usage
+------------|-----------|--------|---------------
+`@id`       | yes       | string | Name of the binding to generate. Will appear in trace log output to identify this binding when loaded.
+`@service`  | yes       | string | Name of the service the generated binding will bind to. This service name must match the ESDL definition's service name.
+
+A `<em:ServiceDefinition>` element contains the ESDL definition of the service.
+
+An `<em:Include>` element will be replaced in the output by the contents of the file it references.
+
+### Including Files
+
+The manifest allows you to organize your ESDL scripts and XSLTs as you see fit in a repo, then use `<em:Include>` tags in the template to include the referenced file in the binding output.
+
+Include Element Attributes
+
+Attribute | Required? | Value  | Usage
+----------|-----------|--------|---------------
+`@file`   | yes       | string | Name of the file to include. If not an an absolute path, all of the -I include paths are searched to find it
+
+The `<em:Include>` operation can be used in these places in the manifest:
```

Comment:
Yes the list is comprehensive. I think you have a good point. I'll update so that its supported anywhere, but I'd like to support recursive includes in a future PR.